azext_edge/tests/edge/orchestration/resources/connector/opcua/int/helpers.py

```diff
@@ -0,0 +1,396 @@
+    retries = 5
+    while retries > 0:
+        try:
+            secret = run(f"az keyvault secret show --vault-name {kv_name} -n {secret_name}")
+            assert secret
+            return
+        except (CLIInternalError, AssertionError):
+            retries -= 1
+            sleep(5)
+    raise AssertionError(f"Secret {secret_name} not found in keyvault {kv_name} or invalid value.")
+
+
+def assert_spc_secret_exists(spc_records: list, spc_name: str, instance_name: str, resource_group: str, cert_file: str):
+    p = Path(cert_file)
+    file_name_info = (p.stem, p.suffix)
+    cert_extension = file_name_info[1].replace(".", "")
+    secret_name = f"{file_name_info[0]}-{cert_extension}"
+    spc_record = next((rec for rec in spc_records if rec["name"] == spc_name), None)
+    assert spc_record
+    objects = spc_record["properties"].get("objects", "")
+    assert secret_name in objects
+    return
+
+
+def assert_ssc_secret_exists(
+    secretsync_records: list,
+    extended_location: str,
+    resource_group: str,
+    cert_file: str,
+    ssc_name: Optional[str],
+):
+    p = Path(cert_file)
+    file_name_info = (p.stem, p.suffix)
+    cert_extension = file_name_info[1].replace(".", "")
+    secret_name = f"{file_name_info[0]}-{cert_extension}"
+    result = next((rec for rec in secretsync_records if rec["name"] == ssc_name), None)
+    assert result is not None, f"SecretSync record '{ssc_name}' not found"
+    assert result["extendedLocation"]["name"] == extended_location
+    assert result["resourceGroup"] == resource_group
+    assert result["name"] == ssc_name
+    secret_mappings = result["properties"].get("objectSecretMapping", [])
+    assert any(mapping.get("sourcePath", "") == secret_name for mapping in secret_mappings)
+
+
+def assert_cluster_side_secret_exists(
+    spc_name: str,
+    secret_sync_name: str,
+    cert_file: str,
+):
+    p = Path(cert_file)
+    secret_name = f"{p.stem}{p.suffix}"
+    secret_value = p.read_bytes()
+    # get the current secret provider class
+    list_result = run_json("kubectl get secretproviderclass -A -o json")["items"]
+    assert list_result
+    spc_data = next(spc for spc in list_result if spc["metadata"]["name"] == spc_name)
+    aio_namespace = spc_data["metadata"]["namespace"]
+    secret_data = run_json(f"kubectl get secret {secret_sync_name} -n {aio_namespace} -o json")
+    assert secret_name in secret_data["data"]
+    # decode the secret value into bytes
+    decoded = b64decode(secret_data["data"][secret_name])
+    assert decoded == secret_value
+
+
+def assert_kv_secret_not_exists(kv_id: str, cert_file: str):
+    kv_name = kv_id.rsplit("/", maxsplit=1)[-1]
+    p = Path(cert_file)
+    file_name_info = (p.stem, p.suffix)
+    cert_extension = file_name_info[1].replace(".", "")
+    secret_name = f"{file_name_info[0]}-{cert_extension}"
+
+    try:
+        run(f"az keyvault secret show --vault-name {kv_name} -n {secret_name}", expect_failure=True)
```

Comment:
@Elsie4ever  JFYI `expect_failure=True` in `run` can be used to indicate that the test command is expected to raise error.